flutter_frontend_server/lib/server.dart

```diff
@@ -169,3 +182,62 @@ Future<int> starter(
   frontend.listenAndCompile(compiler, input ?? stdin, options, completer);
   return completer.future;
 }
+
+/// A [RecursiveVisitor] that replaces [Object.toString] overrides with
+/// `super.toString()`.
+class ToStringVisitor extends RecursiveVisitor<void> {
+  /// The [packageUris] must not be null.
+  ToStringVisitor(this._packageUris) : assert(_packageUris != null);
+
+  /// A set of package URIs to apply this transformer to, e.g. 'dart:ui' and
+  /// 'package:flutter/foundation.dart'.
+  final Set<String> _packageUris;
+
+  /// Turn 'dart:ui' into 'dart:ui', or
+  /// 'package:flutter/src/semantics_event.dart' into 'package:flutter'.
+  String _importUriToPackage(Uri importUri) => '${importUri.scheme}:${importUri.pathSegments.first}';
+
+  @override
+  void visitProcedure(Procedure node) {
+    if (
+      node.name.name        == 'toString' &&
+      node.enclosingClass   != null       &&
+      node.enclosingLibrary != null       &&
+      _packageUris.contains(_importUriToPackage(node.enclosingLibrary.importUri))
+    ) {
+      node.function.replaceWith(
+        FunctionNode(
+          ReturnStatement(
+            SuperMethodInvocation(
+              node.name,
+              Arguments(<Expression>[]),
+            ),
+          ),
+        ),
+      );
+    }
+    super.visitProcedure(node);
```

Comment:
Do we really need to go deeper? Not going beyond members might save a little bit of compilation time.
Also consider adding

  @override
  defaultMember(Member node) {}

Reply:
Done

Reply:
Ah, I missed something here.  I'll ping this again when I push up the change that should fix it.

Reply:
Ok, I removed the super call this time.  I've also added the ability to add an `@pragma` annotation to a `toString` method so we let it through.